src/theme/variables.scss

```diff
@@ -19,26 +24,120 @@ $tertiary: hsl(47, 100%, 92%);
+  $result: (
+    "color-primary": $primary,
+    "color-primary-light": color.change($primary, $lightness: 80%),
+    "color-primary-lighter": color.change($primary, $lightness: 95%),
+    "color-primary-contrast": contrast-colour.choose-contrast-color($primary),
+    "color-secondary": $secondary,
+    "color-secondary-light": color.change($secondary, $lightness: 80%),
+    "color-secondary-lighter": color.change($secondary, $lightness: 95%),
+    "color-secondary-contrast": contrast-colour.choose-contrast-color($secondary),
+    //  GRADIENTS
+    // primary-gradient
+    "gradient-primary-start": color.adjust($primary, $lightness: 33%),
+    "gradient-primary-end": color.adjust($primary, $saturation: 2%, $lightness: 13%),
+    // secondary-gradient
+    "gradient-secondary-start": color.adjust($secondary, $saturation: 4%),
+    "gradient-secondary-end": color.adjust($secondary, $saturation: 5%, $lightness: -19%),
+  );
+
+  @if ($tertiary != null) {
+    $tertiary-map: (
+      "color-tertiary": $tertiary,
+      "color-tertiary-light": color.change($tertiary, $lightness: 80%),
+      "color-tertiary-lighter": color.change($tertiary, $lightness: 95%),
+      "color-tertiary-contrast": contrast-colour.choose-contrast-color($tertiary),
+    );
+    $result: map.merge($result, $tertiary-map);
+  } @else {
+    // Auto generate a tertiary colour map from base colours
+  }
+  @return $result;
+}
+
+$colorPalette: getColorPalette($color-primary, $color-secondary);
+
+@mixin ionColorPalette {
+  --ion-color-primary: #{map-get($colorPalette, "color-primary")};
+  --ion-color-primary-light: #{map-get($colorPalette, "color-primary-light")};
+  --ion-color-primary-lighter: #{map-get($colorPalette, "color-primary-lighter")};
+  --ion-color-primary-contrast: #{map-get($colorPalette, "color-primary-contrast")};
+
+  --ion-color-secondary: #{map-get($colorPalette, "color-secondary")};
+  --ion-color-secondary-light: #{map-get($colorPalette, "color-secondary-light")};
+  --ion-color-secondary-lighter: #{map-get($colorPalette, "color-secondary-lighter")};
+  --ion-color-secondary-contrast: #{map-get($colorPalette, "color-secondary-contrast")};
+
+  --ion-color-tertiary: #{map-get($colorPalette, "color-tertiary")};
+  --ion-color-tertiary-light: #{map-get($colorPalette, "color-tertiary-light")};
+  --ion-color-tertiary-lighter: #{map-get($colorPalette, "color-tertiary-lighter")};
+  --ion-color-tertiary-contrast: #{map-get($colorPalette, "color-tertiary-contrast")};
+
+  //  GRADIENTS
+  // primary-gradient
+  --ion-gradient-primary-start: #{map-get($colorPalette, "gradient-primary-start")};
+  --ion-gradient-primary-end: #{map-get($colorPalette, "gradient-primary-end")};
+  // secondary-gradient
+  --ion-gradient-secondary-start: #{map-get($colorPalette, "gradient-secondary-start")};
+  --ion-gradient-secondary-end: #{map-get($colorPalette, "gradient-secondary-end")};
+}
+
+.theme-default {
+  @include ionColorPalette;
```

Comment:
incredible how quickly a class gets reduced down to a single line of code. Although I'm wondering within this class we should be defining the $primary and $secondary variables, generating the pallets and passing to the mixin. That way everything for a theme can be encapsulated within a single class.

We can also then move that class to a separate folder (e.g. `themes/default.scss`) and use as a template for scaffolding out further themes